src/Desktop.Analyzers/Core/TypesShouldNotExtendCertainBaseTypes.cs

```diff
@@ -31,79 +34,114 @@ public abstract class TypesShouldNotExtendCertainBaseTypesAnalyzer : DiagnosticA
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182171.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(SystemXmlXmlDocumentRule, SystemApplicationExceptionRule, SystemCollectionsCollectionBaseRule, SystemCollectionsDictionaryBaseRule, SystemCollectionsQueueRule, SystemCollectionsReadOnlyCollectionBaseRule, SystemCollectionsSortedListRule, SystemCollectionsStackRule);
 
+        private static readonly Dictionary<string, DiagnosticDescriptor> s_badBaseTypesToDescriptor = new Dictionary<string, DiagnosticDescriptor>
+                                                    {
+                                                        {"System.ApplicationException", SystemApplicationExceptionRule},
+                                                        {"System.Xml.XmlDocument", SystemXmlXmlDocumentRule},
+                                                        {"System.Collections.CollectionBase", SystemCollectionsCollectionBaseRule},
+                                                        {"System.Collections.DictionaryBase", SystemCollectionsDictionaryBaseRule},
+                                                        {"System.Collections.Queue", SystemCollectionsQueueRule},
+                                                        {"System.Collections.ReadOnlyCollectionBase", SystemCollectionsReadOnlyCollectionBaseRule},
+                                                        {"System.Collections.SortedList", SystemCollectionsSortedListRule},
+                                                        {"System.Collections.Stack", SystemCollectionsStackRule},
+                                                    };
+
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterCompilationStartAction(AnalyzeCompilationStart);
+        }
+
+        private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var badBaseTypes = s_badBaseTypesToDescriptor.Keys
+                                .Select(bt => context.Compilation.GetTypeByMetadataName(bt))
+                                .Where(bt => bt != null)
+                                .ToImmutableHashSet();
+
+            context.RegisterSymbolAction((saContext) =>
```

Comment:
Add check for empty `badBaseTypes` before register symbol action, analyzer should bail if so.